.github/workflows/cd.yml

```diff
@@ -0,0 +1,134 @@
+name: Continuous Deployment
+
+on:
+    push:
+        tags:
+        -   'v[0-9]+.[0-9]+.[0-9]+*'
+
+jobs:
+
+    validate-release-tag:
+
+        if: github.repository == 'aiidateam/aiida-quantumespresso'
+        runs-on: ubuntu-latest
+
+        steps:
+        -   name: Checkout source
+        -   uses: actions/checkout@v2
+
+        -   name: Set up Python 3.8
+            uses: actions/setup-python@v2
+            with:
+                python-version: '3.8'
+
+        -   name: Validate the tag version against the package version
+        -   run: python .github/workflows/validate_release_tag.py $GITHUB_REF
+
+    pre-commit:
+
+        needs: [validate-release-tag]
+        runs-on: ubuntu-latest
+
+        steps:
+        -   name: Checkout source
+        -   uses: actions/checkout@v2
+
+        -   name: Cache Python dependencies
+            id: cache-pip
+            uses: actions/cache@v1
+            with:
+                path: ~/.cache/pip
+                key: pip-pre-commit-${{ hashFiles('**/setup.json') }}
+                restore-keys:
+                    pip-pre-commit-
+
+        -   name: Set up Python
+            uses: actions/setup-python@v2
+            with:
+                python-version: '3.8'
+
+        -   name: Install Python dependencies
+            run: pip install -e .[pre-commit,tests]
+
+        -   name: Run pre-commit
+            run: pre-commit run --all-files || ( git status --short ; git diff ; exit 1 )
+
+    tests:
+
+        needs: [validate-release-tag]
+        runs-on: ubuntu-latest
+
+        strategy:
+            matrix:
+                python-version: ['3.6', '3.7', '3.8', '3.9']
+
+        services:
+            postgres:
+                image: postgres:10
+                env:
+                    POSTGRES_DB: test_${{ matrix.backend }}
```

Comment:
We don't set it and we are really just relying on defaults. When you don't explicitly specify `AIIDA_TEST_PROFILE`, calling `pytest` will automatically generate an entire dummy profile with full postgres database cluster on the fly. I would have to update the workflows a bit and test, because I think we can simplify them a lot. I don't think `aiida-quantumespresso` should run the tests against both backends. Anyway Django is getting removed soon.

Reply:
I have updated this to be in line with the cleaned up `ci` workflow. No need to predefine the postgres service.